maintainers/scripts/haskell/transitive-broken-packages.nix

```diff
@@ -0,0 +1,21 @@
+let
+  nixpkgs = import ../../..;
+  inherit (nixpkgs {}) pkgs lib;
```

Comment:
We need to pass a `config` here to make sure we don't get false positives by allowing the following things:

* `allowUnsupportedSystem`
* `allowUnfree`
* `allowInsecurePredicate = _: true;`

Not sure if I missed anything.

Reply:
For unsupported system we kind of need to come up with a solution, since it involves a similar problem as marking stuff as broken unfortunately.

What I've been doing until now is adding reverse dependencies to unsupported systems. The problem is not as bad here as for marking stuff as broken, so I think we can hold off on a “proper” solution for this.

Reply:
Trying to prevent false positives seems really finnicky to me and I am afraid of overlooking something. I prefer sticking with the diff solution from above. It doesn‘t run terribly long and is conceptually simpler.
Note that I think that the code as it is does not have false positives caused by allowUnfree e.g.

Different platforms is a huge issue. I guess the output of this script actually depends on the platform we run this on. I guess in principle we could run this script for the whole matrix allowX and targetPlatform. But I wouldn‘t be sure what to do with the information. A very fine granular config file?

Reply:
>  Note that I think that the code as it is does not have false positives caused by allowUnfree e.g. I think.

That is not true, it has all the issues I mentioned. It is not very noticeable since most packages support `x86_64-linux`. As an example, if you pass `localSystem = "x86_64-linux"` to your expression, it will contain (among other unsupported packages) `Euterpea` which is not broken nor are any of its dependencies.

Reply:
Have you tested that? I am uncertain as to why that would be happening.

We evaluate the package set twice once with allowBroken once without. When `Euterpea` doesn‘t build on x86_64-linux it wouldn‘t exist in both package sets so it shouldn‘t be in their difference either.

I may well be getting this wrong. If so I declare that I am confused and something is not working as I think it is. (Or I have written other code than I thought I wrote?)